history/history-impl/src/main/java/com/duckduckgo/history/impl/HistoryRepository.kt

```diff
@@ -98,6 +102,26 @@ class RealHistoryRepository(
         }
     }
 
+    override suspend fun removeHistoryEntryByUrl(url: String) {
+        withContext(dispatcherProvider.io()) {
+            historyDao.getHistoryEntryByUrl(url)?.let {
```

Comment:
We shouldn't need to query first and then delete. I think we should have a delete query specifying `DELETE FROM history_entries WHERE url = :url`